connection/auth.js

```diff
@@ -192,6 +196,63 @@ class AccessTokenAuthenticator {
   };
 }
 
+export class AuthClientCredentials {
+  constructor(creds) {
+    this.clientSecret = creds.clientSecret;
+  }
+}
+
+class ClientCredentialsAuthenticator {
+  constructor(http, creds, config) {
+    this.http = http;
+    this.creds = creds;
+    this.openidConfig = config;
+  }
+
+  refresh = () => {
+    this.validateOpenidConfig();
+    return this.requestAccessToken()
+      .then(tokenResp => {
+        return {
+          accessToken: tokenResp.access_token,
+          expiresAt: calcExpirationEpoch(tokenResp.expires_in),
+          refreshToken: tokenResp.refresh_token
+        };
+      })
+      .catch(err => {
+        return Promise.reject(
+          new Error(`failed to refresh access token: ${err}`)
+        );
+      });
+  };
+
+  validateOpenidConfig = () => {
+    if (this.openidConfig.provider.token_endpoint
+      .includes("https://login.microsoftonline.com")) {
+      this.openidConfig.scope = this.openidConfig.clientId + "/.default"
+    } else {
+      this.openidConfig.scope = null
+    }
+  };
+
+  requestAccessToken = () => {
+    var url = this.openidConfig.provider.token_endpoint;
+    var params = new URLSearchParams({
+      grant_type: "client_credentials",
+      client_id: this.openidConfig.clientId,
+      client_secret: this.creds.clientSecret,
+    });
+
+    // Azure requires scope, whereas Okta fails if it's provided
+    if (this.openidConfig.scope) {
+      params.append("scope", this.openidConfig.scope)
+    }
```

Comment:
hmmmm, I'm a bit confused by this comment :)
Afaik okta does require a scope for CC ("some_scope" in our case)

Reply:
Okta was failing to provide access token because you can't request an access token with client credentials using the `openid` scope, which is what the metadata endpoint was returning. So setting the scope to null was working for Okta.

I was unaware of the `some_scope` requirement for our Okta server, so after allowing configurable scope, and passing in `some_scope` with `AuthClientCredentials`, it now works. Thanks